clcache.py

```diff
@@ -164,33 +164,43 @@ def clean(self, stats, maximumSize):
         except ImportError:
             walker = os.walk
 
-        objects = [os.path.join(root, "object")
-                   for root, folder, files in walker(self.objectsDir)
-                   if "object" in files]
+        objectInfos = []
+        for x in xrange(16):
+            try:
+                objects = [os.path.join(root, "object")
+                       for root, folder, files in walker(self.objectsDir)
+                       if "object" in files]
+
+                objectInfos = [(os.stat(fn), fn) for fn in objects]
```

Comment:
If it's `stat` which threw, why bother re-trying? And why re-iterate the `object` directory? Wouldn't it be more straightforward to just skip files for which `os.stat` fails, like

```python
objectInfos = []
for o in objects:
    try:
        objectInfos.append((os.stat(fn), fn))
    except:
        pass
```

As for parallel `clcache` processes causing this: I would hope that this cannot be true because there's a lock file to prevent this kind of issue...